src/flow/core/processing.rs

```diff
@@ -84,7 +84,7 @@ pub trait Ingredient
                 data: prelude::Records,
                 domain: &prelude::DomainNodes,
                 states: &prelude::StateMap)
-                -> prelude::Records;
+                -> Option<prelude::Records>;
```

Comment:
Agh, that's going to be annoying to incorporate into `partial-materialization`, but I guess it's necessary here.